src/core/agents.jl

```diff
@@ -221,3 +211,155 @@ macro agent(struct_repr)
         end
     quote Base.@__doc__($(esc(expr))) end
 end
+
+"""
+    @multiagent struct YourParentAgentType{X,Y}(BaseAgentType) [<: OptionalSupertype]
+        @agent FirstAgentType{X}
```

Comment:
I like having `@agent` there, it seems prettier to me :D 

More seriously, there is a somewhat good reason for that (but I really like it more syntactically for the more uniformity with normal agents syntax). The more serious reason is that we could in the future make the multiagent macro work with already defined agents too e.g. with a syntax like so

```julia
@multiagent Animal(Wolf, Hawk)
```

which is useful when some library depends on `Agents`  (e.g. it could be interesting for `MicrobeAgents`) and that library exports different agent types for the final user which can be maybe be mixed by the final user in his model. For this reason too, it seems to me that when instead the agents are defined inside multiagent, they should be created with the macro before them.

But I have to say that I really like the `@agent` macro there in any case, but we could actually change the syntax in

```julia
@multiagent YourMultiAgentType{X,Y}(BaseAgentType) [<: OptionalSupertype] begin
    @agent FirstAgentComponent{X}
        ....
    end
    @agent SecondAgentComponent{Y}
        ....
    end
end
```

which makes a bit clearer that the agents are just components of multiagent. 


Actually a third reason I thought about now is that if and when we will allow multi-space models we could extend the syntax to have different base-types for different agent components and we could attach that to the components:


```julia
@multiagent YourMultiAgentType{X,Y} [<: OptionalSupertype] begin
    @agent FirstAgentComponent{X}(BaseAgent1) 
        ....
    end
    @agent SecondAgentComponent{Y}(BaseAgent2) 
        ....
    end
end
```

Actually we could also make this transition now, it is a little more verbose because for now they should be all equal, but maybe this is already better

Reply:
okay, resolved, we keep the `@agent`.